src/game_api/texture.cpp

```diff
@@ -274,34 +276,40 @@ void reload_texture(const char** texture_name)
 
     auto renderer_ptr = render.renderer();
     auto load_texture = *vtable_find<LoadTextureFunT*>(renderer_ptr, c_LoadTextureVirtualIndex);
+
+    if (texture_name->dx_resource)
+    {
+        texture_name->dx_resource->release(); // should be safe since apparently the objects have reference counter, essentially working like shared pointer
+        game_free(texture_name->dx_resource);
+    }
     load_texture(renderer_ptr, texture_name);
 }
 
 void set_heart_color_from_texture(TEXTURE vanilla_id, TEXTURE custom_id)
 {
     uint32_t char_index = (uint32_t)vanilla_id - 285;
-    if (char_index >= 0 && char_index <= 19)
+    if (char_index > 19)
+        return;
```

Comment:
Only inverted this if check, but the indentation change shows as some bigger difference